src/nvim/lua/vim.lua

```diff
@@ -118,11 +118,21 @@ local function _update_package_paths()
   last_nvim_paths = cur_nvim_paths
 end
 
+local function _os_proc_capsinfo()
+
+  local cmd = { 'xset', 'q', }
```

Comment:
I still think we want a convenient way to override this, if someone makes a function that works for wayland (or bulitin console, input method with soft-capslock etc). It is possible to overwrite `vim._os_proc_capsinfo`, but I think there should be "less friction" compared to say overriding `os_proc_info` (desktop env vary a lot faster than basic os services).

Reply:
Yes, this is true, the functional approach here will be much better than just communicating with shell, there you specified the nvim-x11 but I have to get some time to fully understand its working, I didn't have much exp with wayland now, but if its a need of hour, I can understand it.
Also would like to listen to the view of others too.

Reply:
I didn't say this PR should fix wayland, rather just that this function should look more encouraging for a _user config_ to override. Later on we could do clippoard-style provider detection.

Reply:
Yes its true

Reply:
@bfredl Can that wait until another PR? This one is getting too noisy.

> It is possible to overwrite vim._os_proc_capsinfo

Any underscore-prefix function in `vim` Lua module is not part of public API, so let's not suggest this to any users.

Reply:
@justinmk I thought I _very explictly_ stated that more flexible detection should wait until later. The change I have in mind could be as simple as renaming the function for now (and document it)

> any underscore-prefix function in vim Lua module is not part of public API, so let's not suggest this to any users.

I agree with the premise (underscore functions sholudn't be changed), but the conclusion is quite backwards. If users need to override a "underscore-prefix function in vim Lua module" to properly use a new feature, isn't the obvious conclusion that the function should be renamed?

Reply:
> If some users need to override a "underscore-prefix function in vim Lua module" to properly use a new feature, isn't the obvious conclusion that the function should at least be renamed?

I don't know. What I do know is that underscore-prefixed Lua functions may be removed/renamed in the future and there must not be even the slightest implication that backwards-compatibility will be supported for them in userspace.

Reply:
I don't think we should tie backwards-compatibility to underscore. Worst case in this case is just that the override has no effect. When user ask on gitter how to make soft-capslock (or linux console, as @ZyX-I just mentioned) to work, I don't want to need to say "In general you should not change underscore functions, but in this case you should change `vim._os_proc_capsinfo` for now" (lying to them and saying it is impossible is not an option).

Reply:
Also BTW, shouldn't `proc` be dropped from the name?